client/src/main/java/org/apache/celeborn/client/read/CelebornInputStream.java

```diff
@@ -479,6 +481,7 @@ private PartitionReader createReaderWithRetry(
                 fetchChunkMaxRetry,
                 location,
                 e);
+            location = location.getPeer();
```

Comment:
Why change location to peer at this position?

Reply:
@r7raul1984, does the changed location have the corresponding shuffle data to read?

Reply:
> Since the previous location has already been excluded, I think we need to try fetching from peers.

Should send bufferStreamEnd first before change location to peer

Reply:
> > Since the previous location has already been excluded, I think we need to try fetching from peers.
> 
> Should send bufferStreamEnd first before change location to peer

yes, you are right